### PR TITLE
Remove CGUIWindowLoginScreen::LoadProfile()

### PR DESCRIPTION
**Pull request by garbear**
This PR removes the static `LoadProfile()` function. It consolidates all dependencies into four functions in CProfilesManager.

The logic has been preserved, but it was scattered all around to begin with, so there is probably issues.

## Motivation and Context
For #13412

The next step is to invert the dependencies, and have external services register themselves with CProfilesManager. Then we make CPeripherals one of these services so that it can be profile-aware and load the proper peripheral settings.

## How Has This Been Tested?
Brief testing revealed only the known peripherals at login bug.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -1054,9 +1053,12 @@ bool CApplication::Initialize()
 
   StartServices();
 
+  // GUI depends on seek handler
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: please elaborate

File: xbmc/profiles/ProfilesManager.cpp [hunk 2]
```diff
@@ -314,10 +361,108 @@ bool CProfilesManager::LoadProfile(size_t index)
...
+  // restart PVR services
+  pvrManager.Init();
+
+  favouritesManager.ReInit(GetProfileUserDataFolder());
+
+  serviceAddons.Start();
+
+  g_application.UpdateLibraries();
+
+  stereoscopicsManager.Initialize();
```
Comment (FernetMenta) [inline xbmc/profiles/ProfilesManager.cpp]: stereoscopic manager is owed by GUIComponent. Its init method shold not be caled from outside

File: xbmc/Application.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/Application.cpp]: Search code for "GetSeekHandler". There's a bunch of places that call into the seek handler, including from GUI code.

File: xbmc/profiles/ProfilesManager.cpp (hunk 2 above)
Comment (garbear) [reply xbmc/profiles/ProfilesManager.cpp]: every single line in this function should not be called here. the next step is removing them all

File: xbmc/Application.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/Application.cpp]: I just moved existing code. Git history might reveal more: https://github.com/garbear/xbmc/commit/8b841c3

**Comment by garbear** [comment]
good to go?

**Comment by FernetMenta** [comment]
as far as I am concerned, yes

**Pull request merged**